src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -334,6 +337,11 @@ CompleteShardPlacementTransactions(XactEvent event, void *arg)
 	shardConnectionHash = NULL;
 	XactModificationLevel = XACT_MODIFICATION_NONE;
 	subXactAbortAttempted = false;
+
+	if (MultiShardCommitProtocol == COMMIT_PROTOCOL_BARE)
+	{
+		MultiShardCommitProtocol = SavedMultiShardCommitProtocol;
+	}
```

Comment:
I don't think it's quite that unlikely. What if we add infrastructure - and we should - to propage GUCs to workers?  What if your task gets interrupted? At the moment you'll leak the BARE setting in that case afaics.